### PR TITLE
Add config options to restrict signup on certain platforms

### PR DESCRIPTION
**Pull request by Zaprit**
Due to a recent increate in abuse of our platform, I have implemented an option to disallow initial signup from certain platforms. It defaults to allow all signup attempts but can be easily restricted in the config. Also fixes the Qodana warnings in LoginController.cs

### CODE REVIEW

File: ProjectLighthouse.Servers.GameServer/Controllers/Login/LoginController.cs [hunk 1]
```diff
@@ -57,27 +49,27 @@ public async Task<IActionResult> Login()
...
 
-        if (username == null)
+        if (String.IsNullOrEmpty(username))
         {
             Logger.Warn("Unable to determine username, rejecting login", LogArea.Login);
             return this.Forbid();
         }
 
-        await this.database.RemoveExpiredTokens();
+        await database.RemoveExpiredTokens();
```
Comment (sudokoko) [inline ProjectLighthouse.Servers.GameServer/Controllers/Login/LoginController.cs:60]: Doesn't using the primary constructor DbContext variable cause a warning? I forget how it works exactly in LH but in my own personal projects it causes a warning.

File: ProjectLighthouse.Servers.GameServer/Controllers/Login/LoginController.cs [hunk 2]
```diff
@@ -18,15 +17,8 @@ namespace LBPUnion.ProjectLighthouse.Servers.GameServer.Controllers.Login;
...
-public class LoginController : ControllerBase
+public class LoginController(DatabaseContext database) : ControllerBase
 {
-    private readonly DatabaseContext database;
-
-    public LoginController(DatabaseContext database)
-    {
-        this.database = database;
-    }
-
```
Comment (sudokoko) [inline ProjectLighthouse.Servers.GameServer/Controllers/Login/LoginController.cs:29]: Didn't we want to hold off on using primary constructors? If I recall correctly there was a PR that we removed those changes from.

File: ProjectLighthouse/Configuration/ConfigurationCategories/AuthenticationConfiguration.cs [hunk 3]
```diff
@@ -5,4 +5,9 @@ public class AuthenticationConfiguration
     public bool RegistrationEnabled { get; set; } = true;
     public bool AutomaticAccountCreation { get; set; } = true;
     public bool VerifyTickets { get; set; } = true;
+
+    public bool AllowRPCNSignup { get; set; } = true;
+
+    public bool AllowPSNSignup { get; set; } = true;
```
Comment (sudokoko) [inline ProjectLighthouse/Configuration/ConfigurationCategories/AuthenticationConfiguration.cs:11]: Increment base configuration version number

**Review by sudokoko — CHANGES_REQUESTED** [review]

File: ProjectLighthouse.Servers.GameServer/Controllers/Login/LoginController.cs (hunk 2 above)
Comment (Zaprit) [reply ProjectLighthouse.Servers.GameServer/Controllers/Login/LoginController.cs:29]: I don't know, I was just listening to rider. I can revert the change if need be.

File: ProjectLighthouse.Servers.GameServer/Controllers/Login/LoginController.cs [hunk 4]
```diff
@@ -130,6 +130,23 @@ public async Task<IActionResult> Login()
...
+            // Block RPCN signups if forbidden in config
+            if (npTicket.Platform == Platform.RPCS3 && !ServerConfiguration.Instance.Authentication.AllowRPCNSignup)
+            {
+                Logger.Warn(
+                    $"New user tried to sign up via RPCN, and that is forbidden in the config, username={username}, remoteIpAddress={remoteIpAddress}",
+                    LogArea.Login);
+            }
+
+            // Block PSN signups if forbidden in config
+            if (npTicket.Platform == Platform.RPCS3 && !ServerConfiguration.Instance.Authentication.AllowPSNSignup)
```
Comment (FeTetra) [inline ProjectLighthouse.Servers.GameServer/Controllers/Login/LoginController.cs]: Shouldn't this be checking for `Platform.PS3` `Platform.Vita`, and `Platform.PSP`

<sub><sup>oops</sup></sub>

Comment (Zaprit) [reply ProjectLighthouse.Servers.GameServer/Controllers/Login/LoginController.cs]: Yeah, I fixed that previously, it now refers to an extension function of the platform enum that checks if it's PSN or not

File: ProjectLighthouse.Servers.GameServer/Controllers/Login/LoginController.cs [hunk 5]
```diff
@@ -21,12 +20,11 @@ namespace LBPUnion.ProjectLighthouse.Servers.GameServer.Controllers.Login;
 public class LoginController : ControllerBase
 {
     private readonly DatabaseContext database;
-
     public LoginController(DatabaseContext database)
     {
         this.database = database;
     }
-
+
```
Comment (Slendy) [inline ProjectLighthouse.Servers.GameServer/Controllers/Login/LoginController.cs:27]:
```suggestion

```

File: ProjectLighthouse.Servers.GameServer/Controllers/Login/LoginController.cs [hunk 6]
```diff
@@ -72,12 +70,12 @@ public async Task<IActionResult> Login()
         switch (npTicket.Platform)
         {
             case Platform.RPCS3:
-                user = await this.database.Users.FirstOrDefaultAsync(u => u.LinkedRpcnId == npTicket.UserId);
+                user = await database.Users.FirstOrDefaultAsync(u => u.LinkedRpcnId == npTicket.UserId);
```
Comment (Slendy) [inline ProjectLighthouse.Servers.GameServer/Controllers/Login/LoginController.cs:73]:
```suggestion
                user = await this.database.Users.FirstOrDefaultAsync(u => u.LinkedRpcnId == npTicket.UserId);
```
I'm down to ditch this style in the future but for now lets just match the rest of the codebase

**Review by Slendy — CHANGES_REQUESTED** [review]

**Review by sudokoko — APPROVED** [review]
See slendy's review, I have nothing to add to it

**Review by Slendy — APPROVED** [review]

**Comment by Zaprit** [comment]
Finally

**Pull request merged**